src/App.tsx

```diff
@@ -17,18 +63,46 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onFilterChange={setFilterCondition}
+                currentFilter={filterCondition}
+                onQueryChange={setQuery}
+                currentQuery={query}
+              />
             </div>
 
+            {!isLoadedTodos && <Loader />}
+
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!isLoadingError
+                ? (
+                  <TodoList
+                    todos={filteredTodos}
+                    selectedTodo={selectedTodo}
+                    isTodoInfoRequested={isTodoInfoRequested}
+                    onTodoInfoRequest={setIsTodoInfoRequested}
+                    onUserIdChange={setUserId}
+                    onSelectedTodoChange={setSelectedTodo}
+                  />
+                )
+                : (
+                  <div className="message p-2 my-4 is-danger">
+                    <p className="message-body">Error loading goods</p>
+                  </div>
+                )}
+
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {isTodoInfoRequested && (
```

Comment:
You can render your `TodoModal` when you have the selected todo

```suggestion
      {selectedTodo && (
```